addons/website_sale/controllers/main.py

```diff
@@ -1227,3 +1232,61 @@ def get_combination_info_website(self, product_template_id, product_id, combinat
             })
         res['carousel'] = carousel_view
         return res
+
+    # ------------------------------------------------------
+    # Products Search Bar
+    # ------------------------------------------------------
+
+    @http.route('/shop/products/autocomplete', type='json', auth='public', website=True)
+    def products_autocomplete(self, term, limit, product_options, **kwargs):
+        """
+        Returns list of products according to the term and product options
+
+        Params:
+            term (str): search term written by the user
+            limit (int): number of products we want to return in the list
+            product_options (dict): product options set by the user (from snippet option)
+
+        Returns:
+            {} (dict): list of products
+        """
+        display_description = product_options['display_description']
+        display_price = product_options['display_price']
+        order_by = product_options['order_by']
+
+        domain = self._get_search_domain(term, None, None, display_description)
+        products = request.env['product.template'].search(domain, limit=limit, order=order_by)
+        pricelist = request.website.get_current_pricelist()
+
+        def get_product_info(product):
+            product_info = {
+                'id': product.id,
+                'name': product.name,
+                'url': product.website_url,
+            }
+
+            if display_description:
+                product_info['description'] = product.description_sale
+
+            if display_price:
+                first_possible_combination = product._get_first_possible_combination()
+                combination_info = product._get_combination_info(first_possible_combination,
+                                                                 add_qty=1,
+                                                                 pricelist=pricelist)
+                product_info.update({
+                    'price': combination_info['price'],
+                    'currency_symbol': pricelist.currency_id.symbol,
+                    'currency_position': pricelist.currency_id.position,
+                })
+
+            return product_info
+
+        products_info = [get_product_info(product) for product in products]
+
+        order_by_info = order_by.split(' ')
```

Comment:
this part some superfluous.

You will sort the product into the current list, but it is possible to have cheapeast product on the next part.
So it is simple imo to stay coherent and always say that with sort on catalog price and not the first combinaison price.


and in which case you will have a multi order_by ?

Reply:
I agree, if you make an SQL query, the sort should be included on the query, not done after in python, otherwise it's not going to work with any kind of pager using limit, offset.

Also currently you return the "price" (computed, depending on pricelist rules etc) but you name the order key "list_price" which is another field (the calatog price). So either call it list_price and sort by list_price, or if we sort by price, name it price then.

Reply:
There is no multi order_by, the split is to get asc / desc. Not sure what you want after your explanations... x) Should I just sort on list_price and not caring that the displayed price will not be displayed? Or should we forget about the first combination stuff and display the list_price ?